app/models/provider_user.rb

```diff
@@ -66,4 +67,11 @@ def can_manage_organisations?
   def downcase_email_address
     self.email_address = email_address.downcase
   end
+
+  def update_notification_preferences
+    if saved_change_to_attribute?(:send_notifications)
+      self.notification_preferences ||= ProviderUserNotificationPreferences.create!(provider_user: self)
+      notification_preferences.update_all_preferences(send_notifications)
```

Comment:
unrelated to this PR but might impact performance: can we amend the implementation of `update_all_preferences` in `ProviderUserNotificationPreferences` to do one update instead of the number of notification preferences?

Reply:
This is outdated now as I've gone for a discreet service called from relevant controllers.